feat: Add jdbc/hive catalogs for iceberg sink. (#1978)
risingwavelabs/risingwave-docs: docs/guides/sink-to-iceberg.md

| s3.secret.key   | Required. Secret key of the S3 compatible object store.|
| database.name   | Required. The database of the target Iceberg table.|
| table.name      | Required. The name of the target Iceberg table.|

> | catalog.name    | Conditional. The name of Iceberg catalog. Can be omitted for storage catalog, required for other catalogs.|

Reviewer: ```suggestion
| catalog.name    | Conditional. The name of the Iceberg catalog. It can be omitted for storage catalog but required for other catalogs.|
```

- Storage catalog: The Storage catalog stores all metadata in the underlying file system, such as Hadoop or S3. Currently, we only support S3 as the underlying file system.
- REST catalog: RisingWave supports the [REST catalog](https://iceberg.apache.org/concepts/catalog/#decoupling-using-the-rest-catalog), which acts as a proxy to other catalogs like Hive, JDBC, and Nessie catalog. This is the recommended approach to use RisingWave with Iceberg tables.

> - Hive catalog: RisingWave supports hive catalog. You need to set `catalog.type` to `hive` to use it.. See [this config](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/hive/config.ini) for full example.

Reviewer: ```suggestion
- Hive catalog: RisingWave supports the Hive catalog. You need to set `catalog.type` to `hive` to use it. See the full example in this [configuration file](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/hive/config.ini).
```

> - Jdbc Catalog: RisingWave supports [jdbc catalog](https://iceberg.apache.org/docs/1.5.0/jdbc/#configurations). See [this config](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/jdbc/config.ini) for full example.

Reviewer: ```suggestion
- Jdbc Catalog: RisingWave supports the [JDBC catalog](https://iceberg.apache.org/docs/latest/jdbc/#configurations). See the full example in this [configuration file](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/jdbc/config.ini).
```
